libs-scala/doobie-slf4j/src/main/scala/com/daml/doobie/logging/Slf4jLogHandler.scala

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.doobie.logging
+
+import org.slf4j.{Logger, LoggerFactory}
+import doobie.util.log.{ExecFailure, LogHandler, ProcessingFailure, Success}
+
+object Slf4jLogHandler {
+
+  def apply(clazz: Class[_]): LogHandler =
+    apply(LoggerFactory.getLogger(clazz))
+
+  def apply(logger: Logger): LogHandler =
+    LogHandler {
+      case Success(s, a, e1, e2) =>
+        logger.debug(s"""Successful Statement Execution:
+                        |
+                        |  ${s.linesIterator.dropWhile(_.trim.isEmpty).mkString("\n  ")}
```

Comment:
If `|` occurs in any of the interpolated data it will be `stripMargin`ed too; it might be better to not use `stripMargin` here at all.

Reply:
Ohhh, this is interesting, thanks.

Failing example.

```
scala> :paste
// Entering paste mode (ctrl-D to finish)

val trololo = "   | trololo"
s"""this
   |  is
   |  an
   |  $trololo
   |  experiment
   |""".stripMargin

// Exiting paste mode, now interpreting.

trololo: String = "   | trololo"
res0: String =
"this
  is
  an
 trololo
  experiment
"
```